clients/gax/lib/google_api/gax/connection.ex

```diff
@@ -24,7 +24,7 @@ defmodule GoogleApi.Gax.Connection do
       @scopes unquote(Keyword.get(opts, :scopes, []))
 
       # Add any middleware here (authentication)
-      plug(Tesla.Middleware.BaseUrl, unquote(Keyword.get(opts, :base_url)))
+      plug(Tesla.Middleware.BaseUrl, Application.get_env(unquote(Keyword.get(opts, :otp_app)), :base_url, unquote(Keyword.get(opts, :base_url))))
```

Comment:
Is the intent for the `:otp_app` keyword arg to be required?

Reply:
Yes, it's added to the template. It should contain the name of the library like `:google_api_pub_sub` so you can set your `config.exs`:

```elixir
config :google_api_pub_sub, :base_url, "example.com"
# OR
config :google_api_pub_sub, :base_url, System.get_env("SOME_ENV_VARIABLE")

```

Reply:
When compiled, this line should become:

```elixir
plug(Tesla.Middleware.BaseUrl, Application.get_env(:google_api_pub_sub, :base_url, "https://pubsub.googleapis.com"))
```